segmenttree/segmenttree.go

```diff
@@ -0,0 +1,8 @@
+package segmenttree
+
+type SegmentTree interface {
```

Comment:
If we implement bulk-loading we would extend this interface with `BulkInsert(values []ValueIntervalTuple)`